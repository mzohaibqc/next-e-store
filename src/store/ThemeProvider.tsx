'use client';
import { ThemeProvider } from 'next-themes';
import { ThemeProviderProps } from 'next-themes/dist/types';

export function Provider(props: ThemeProviderProps) {
  return (
    <ThemeProvider {...props} attribute="class">
      {props.children}
    </ThemeProvider>
  );
}
