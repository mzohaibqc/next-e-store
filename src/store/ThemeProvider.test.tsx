import { render } from '@testing-library/react';
import { expect, test } from 'vitest';
import { Provider as ThemeProvider } from './ThemeProvider';

test('ThemeProvider should render without error', async () => {
  const { container } = render(
    <ThemeProvider>
      <p></p>
    </ThemeProvider>
  );
  // save snapshot
  expect(container).toMatchSnapshot();
});
