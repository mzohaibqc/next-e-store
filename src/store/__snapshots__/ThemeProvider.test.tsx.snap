// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ThemeProvider should render without error 1`] = `
<div>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
  </script>
  <p />
</div>
`;
